src/DIPS.Xamarin.UI/Controls/Modality/ModalityLayout.xaml.cs

```diff
@@ -148,9 +148,15 @@ public void Show(IModalityHandler modalityHandler, View view, Constraint? xConst
             ShowOverlay();
 
             relativeLayout.Children.Add(view, xConstraint, yConstraint, widthConstraint, heightConstraint);
+            view.SizeChanged += ContentSizeChanged;
+        }
+
+        private void ContentSizeChanged(object sender, EventArgs e)
+        {
+            InvalidateLayout();
+            relativeLayout.InvalidateLayout();
```

Comment:
Instead of creating a new relativelayout, try running : `relativeLayout.ForceLayout();`